### ISSUE TITLE
Cannot actually convert anything to SASS map

### ISSUE THREAD

**Issue by iamchristopher** [sass-eyeglass/node-sass-utils#3]
I may just be misunderstanding the documentation (although there is a large character count, it is not obvious how to use this library), I cannot actually get any values converted to a SASS map.

I have a project requirement to inject some JS objects into my SASS file. I have tried many libraries, including this one several times.

[According to the documentation][1], I should be able to cast a given value to its SASS equivalent but everything just returns an empty object:

```
// returns {}
sassUtils.castToSass({ key : 'value' });
```

```
// returns {}
sassUtils.castToSass(true);
```

I also tried doing something like the following:
```
new sassUtils.SassJsMap({ key : 'value' });
```

.. which results in `TypeError: undefined is not a function` pointing to the `map.forEach(callback);` on line 14 of `sass_js_map.js`.

_Is this library actually capable of what I need to do?_ And if so, may I have some guidance as to what I'm doing incorrectly?

[1]: https://github.com/sass-eyeglass/node-sass-utils#sassutilscasttosassjsvalue

**Comment by scottdavis** [comment]
If you look in the [Readme](https://github.com/sass-eyeglass/node-sass-utils#sassutilscasttosassjsvalue) you will see that you will need to make a SassJsMap and pass it into the function to be cast to sass. If you look in the test cases you can see usage [examples](https://github.com/sass-eyeglass/node-sass-utils/blob/master/test/test_sass_js_map.js#L58-L60)

**Comment by iamchristopher** [comment]
That same documentation also says it can take an object and cast it. Is this incorrect?

```
object - cast to sass map where the keys are strings and the values are cast recursively.
```

**Comment by eoneill** [comment]
Note that you're getting back a `sass.type` instance, so `console.log`ing it won't show you anything but an empty object `{}`

```js
var sassMap = sassUtils.castToSass({ key: 'value' });
console.log(sassMap); // {}
console.log(sassUtils.sassString(sassMap)); // (key: value)

var sassBool = sassUtils.castToSass(true);
console.log(sassBool); // {}
console.log(sassUtils.sassString(sassBool)); // true
```

If you wanted to interact with the actual object/values, you'd have to use the `node-sass` methods exposed on the instance.
```js
// inspecting the constructor prototype will show you all of the methods available
console.log(sassMap.constructor.prototype);
for (var i = 0, len = sassMap.getLength(); i < len; i++) {
  var key = sassUtils.sassString(sassMap.getKey(i));
  var value = sassUtils.sassString(sassMap.getValue(i));
  console.log(key, value);
}
```
Also, see [the `node-sass` docs about types](https://github.com/sass/node-sass#typesnumbervalue--unit--)

But that's kinda the whole point of `node-sass-utils` is that you get a more native JS like experience and don't have to fumble around too much with the sass type interface.

Basically, you do whatever operations you need to do in JS, then at the last moment before passing it back into the sass world you call `castToSass`.

Let us know if this addresses your concern.

**Comment by iamchristopher** [comment]
There's where I made my mistake, I didn't realize I had to do the two step process to cast to a SASS map and then use the util method to get the string. Thank you for pointing that out, @eoneill, and thank you @scottdavis as well.
